doc/source/train/user_guide.rst

```diff
@@ -449,105 +404,82 @@ Example: Logging to MLflow and TensorBoard
-    $ cd <TRAINER_RUN_DIR>
-
-    # View the MLflow UI.
-    $ mlflow ui
-
-    # View the tensorboard UI.
-    $ tensorboard --logdir .
-
-
 .. _train-custom-callbacks:
 
 Custom Callbacks
 ++++++++++++++++
 
 If the provided callbacks do not cover your desired integrations or use-cases,
-you may always implement a custom callback by subclassing ``TrainingCallback``. If
-the callback is general enough, please feel welcome to `add it <https://docs
-.ray.io/en/master/getting-involved.html>`_ to the ``ray``
-`repository <https://github.com/ray-project/ray>`_.
+you may always implement a custom callback by subclassing ``Callback``. If
+the callback is general enough, please feel welcome to :ref:`add it <getting-involved>`
+to the ``ray`` `repository <https://github.com/ray-project/ray>`_.
 
 A simple example for creating a callback that will print out results:
 
 .. code-block:: python
 
-    from ray import train
-    from ray.train import Trainer, TrainingCallback
     from typing import List, Dict
 
-    class PrintingCallback(TrainingCallback):
-        def handle_result(self, results: List[Dict], **info):
-            print(results)
+    from ray.air import session
+    from ray.air.config import RunConfig
+    from ray.train.torch import TorchTrainer
+    from ray.tune.logger import LoggerCallback
+
+    # LoggerCallback is a higher level API of Callback.
+    class LoggingCallback(LoggerCallback):
+        def __init__(self) -> None:
+            self.results = []
+
+        def log_trial_result(self, iteration: int, trial: "Trial", result: Dict):
+            self.results.append(trial.last_result)
 
     def train_func():
         for i in range(3):
-            train.report(epoch=i)
+            session.report({"epoch": i})
 
-    trainer = Trainer(backend="torch", num_workers=2)
-    trainer.start()
-    result = trainer.run(
+    callback = LoggingCallback()
+    trainer = TorchTrainer(
         train_func,
-        callbacks=[PrintingCallback()]
+        run_config=RunConfig(callbacks=[callback]),
+        scaling_config=dict(num_workers=2),
     )
-    # [{'epoch': 0, '_timestamp': 1630471763, '_time_this_iter_s': 0.0020279884338378906, '_training_iteration': 1}, {'epoch': 0, '_timestamp': 1630471763, '_time_this_iter_s': 0.0014922618865966797, '_training_iteration': 1}]
-    # [{'epoch': 1, '_timestamp': 1630471763, '_time_this_iter_s': 0.0008401870727539062, '_training_iteration': 2}, {'epoch': 1, '_timestamp': 1630471763, '_time_this_iter_s': 0.0007486343383789062, '_training_iteration': 2}]
-    # [{'epoch': 2, '_timestamp': 1630471763, '_time_this_iter_s': 0.0014500617980957031, '_training_iteration': 3}, {'epoch': 2, '_timestamp': 1630471763, '_time_this_iter_s': 0.0015292167663574219, '_training_iteration': 3}]
-    trainer.shutdown()
+    trainer.fit()
 
-..
-    Advanced Customization
-    ~~~~~~~~~~~~~~~~~~~~~~
+    print("\n".join([str(x) for x in callback.results]))
```

Comment:
Oh this is interesting... is this an intended pattern where callbacks are stateful and accessed after training?